### ISSUE TITLE
Add [NoSnapshot] attribute to avoid sharing a state to other clients when connecting

### ISSUE THREAD

**Issue by bub-bl** [Facepunch/sbox-issues#6928]
### What can't you do?

When a player is connecting to the server, a snapshot of all gameobjects are sent to the new client.
If I make a menu, network them, and open this menu on the host, the menu will be opened on the new client too if the `IsOpen` property have the `[Property]` attribute.

### How would you like it to work?

The idea is to add a new attribute called `[NoSnapshot]` for property. If we add this attribute on a property marked with the `[Property]` attribute, no snapshot will be created for this property. (By default, using `[Property]` attribute snapshot the property to other client)

With that, we can have a property editable in the editor, but without snapshoting them on other clients

### What have you tried?

Above

**Comment by DoctorGurke** [comment]
Why are you networking a menu?

**Comment by bub-bl** [comment]
> Why are you networking a menu?

In some cases, that can be usefull. In my case, im doing a phone and each player have a phone. A lot of state are synced, and i have a lot of RPC to.

Im pretty sure that can be usefull for a lot of people

**Comment by GavrilovNI** [comment]
related #7883

**Comment by GavrilovNI** [comment]
Good idea is to also add [SnapshotOnly] or something like that, so it serializes only for snapshot.
